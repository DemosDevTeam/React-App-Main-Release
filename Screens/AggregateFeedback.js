import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Button,
  TouchableOpacity,
  TextInput,
  Alert,
  ScrollView,
  TouchableHighlight,
  Image
} from 'react-native';
import {firebaseApp} from '../App'
import VideoComponent from '../mainFeedComponents/videoComponent';

export default class AggregateFeedback extends Component<{}>{
  userRef = firebaseApp.database().ref('/Users/' + this.props.navigation.state.params.emailHashAggregateFeedback + "/");
  emailHashMain = this.props.navigation.state.params.emailHashAggregateFeedback;
  videosArr = [];

  componentWillMount() {
    this.setState({loading: true});
    //feedback will be 2d array with each first feedback[i] corresponding to a content piece reacted to by user
    /*for each element (i) of feedback, values are as follows:
      feedback[i][0] = video name
      feedback[i][1] = reaction type
      feedback[i][2] = video pic
      feedback[i][3] = video url
      feedback[i][4] = video name
    */
    var feedback = [];
    this.userRef.child("Reactions").once("value").then((snap) => {
      snap.forEach((child) => {
        var nameOfVideo = child.key;
        var reaction = child.val().reaction;
        var elementArr = [nameOfVideo, reaction];
        feedback.push(elementArr);
      })
    }).then(() => {
      firebaseApp.database().ref('/videos/').once("value").then((snap) => {
        for(var i=0; i<feedback.length; i++){
          if(snap.hasChild(feedback[i][0])){
            var urlpic = snap.child(feedback[i][0]).val().urlpic;
            var urlvideo = snap.child(feedback[i][0]).val().urlvideo;
            var name = snap.child(feedback[i][0]).val().name;

            feedback[i].push(urlpic);
            feedback[i].push(urlvideo);
            feedback[i].push(name);
          }
        }
      }).then(() => {
        for(var i=0; i<feedback.length; i++){
          console.log("added a video");
          console.log(feedback[i]);
          this.videosArr.push(
            <VideoComponent navigation={this.props.navigation} videoUrl={feedback[i][3]} picUrl={feedback[i][2]} videoName={feedback[i][4]} emailHash={this.emailHashMain}/>
          )
        }
      }).then(() => {
        this.setState({loading: false});
      })
    })
  }

  goBack = () => {
    this.props.navigation.navigate('MainFeed', {emailhashmain: this.emailhashmain});
  }

  render() {
    console.disableYellowBox = true;
    if (this.state.loading) {
      return (<Text>Loading...</Text>)
    }
    return (
       <View style={{flex: 1}}>
      <View style={styles.stepzTop}>
        <View style={{flex:1}}>
          <View style={styles.pickContainerz}>

            <TouchableHighlight onPress={this.logout}  style={{flex:1, marginRight: 75}}>
              <View style={styles.pickWrapperz}>
                <View style={styles.circlesTop}>
                  <Image
                    source={{uri: 'https://user-images.githubusercontent.com/18129905/37560140-6b94e5a2-2a09-11e8-9705-6fbb8681a2f2.png'}}
                    style={styles.topIcons}
                  />
                </View>
              </View>
            </TouchableHighlight>


            <Image
              style={{width: 57, height: 40, marginTop: 20}}
              source={{uri: 'https://user-images.githubusercontent.com/18129905/35842080-0e87b16e-0ace-11e8-9fc0-151043ca61fe.png'}}
            />


            <TouchableHighlight onPress={this.updateProfile}  style={{flex:1, marginLeft: 75}}>
              <View style={styles.pickWrapperz}>
                <View style={styles.circlesTop}>
                  <Image
                    source={{uri: 'https://user-images.githubusercontent.com/18129905/37560135-5cd45bd8-2a09-11e8-8af5-fd98c2900ae8.png'}}
                    style={styles.topIcons}
                  />
                </View>
              </View>
            </TouchableHighlight>
          </View>
        </View>
      </View>





      <ScrollView>
         <View style={styles.space}></View>
         <View style={styles.space}></View>
         <Text style={{fontSize: 14, textAlign: 'center'}}>This page allows for you to view your personal feedback for previously viewed videos.</Text>
         <View style={styles.space}></View>
         <View style={styles.container}>{this.videosArr}</View>
         <View style={styles.space}></View>
	 <View style={styles.container}>
         <View style={styles.buttonzRed}>
           <TouchableOpacity onPress={this.goBack}>
             <Text style={{fontSize: 16, textAlign: 'center'}}>Back</Text>
           </TouchableOpacity>
         </View>
         </View>
	 <View style={styles.space2}></View>
	 </ScrollView>





      <View style={styles.stepz}>
        <View style={{flex:1}}>
          <View style={styles.pickContainerz}>
            <TouchableHighlight onPress={this.goToCouncil} style={{flex:1}}>
              <View style={styles.pickWrapperz}>
                <View style={styles.circlesTwo}>
                  <Image
                    source={{uri: 'https://user-images.githubusercontent.com/18129905/37560127-4e64b318-2a09-11e8-9a2e-5d16e6241b7a.png'}}
                    style={styles.arrowWinz}
                  />
                </View>
              </View>
            </TouchableHighlight>

            <TouchableHighlight onPress={this.updateProfile}  style={{flex:1}}>
              <View style={styles.pickWrapperz}>
                <View style={styles.circlesTwo}>
                  <Image
                    source={{uri: 'https://user-images.githubusercontent.com/18129905/37560135-5cd45bd8-2a09-11e8-8af5-fd98c2900ae8.png'}}
                    style={styles.arrowDrawz}
                  />
                </View>
              </View>
            </TouchableHighlight>

            <Image
              style={{width: 100, height: 100, marginTop: 10, marginBottom: 20}}
              source={{uri: 'https://user-images.githubusercontent.com/18129905/35187343-734d21b4-fdf0-11e7-8799-761570dea412.png'}}
            />

            <TouchableHighlight onPress={this.updateFeedback}  style={{flex:1}}>
              <View style={styles.pickWrapperz}>
                <View style={styles.circlesTwo}>
                  <Image
                    source={{uri: 'https://user-images.githubusercontent.com/18129905/37560114-018fe792-2a09-11e8-91d3-22d8f6e49c82.png'}}
                    style={styles.arrowDrawz}
                  />
                </View>
              </View>
            </TouchableHighlight>

            <TouchableHighlight onPress={this.logout}  style={{flex:1}}>
              <View style={styles.pickWrapperz}>
                <View style={styles.circlesTwo}>
                  <Image
                    source={{uri: 'https://user-images.githubusercontent.com/18129905/37560140-6b94e5a2-2a09-11e8-9705-6fbb8681a2f2.png'}}
                    style={styles.arrowWinz}
                  />
                </View>
              </View>
            </TouchableHighlight>
          </View>
        </View>
      </View>
      </View>
    )
  }
}

var styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  images: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    flex: 0,
    marginBottom: 5,
  },
  space: {
    height: 2,
  },
  space: {
    height: 30,
  },
  stepzTop: {
    backgroundColor: '#49C7E3',
    borderRadius: 1,
    flex: 1,
    marginLeft: 0,
    marginRight: 0,
    marginBottom: 0,
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 25,
    paddingBottom: 25,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowRadius: 3,
    shadowOpacity: 0.2,
    shadowColor: 'black',
    textAlign: 'center',
  },
  stepz: {
    backgroundColor: '#ffffff',
    borderRadius: 1,
    flex: 1,
    marginLeft: 0,
    marginRight: 0,
    marginBottom: 0,
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 25,
    paddingBottom: 35,
    shadowOffset: {
      width: 0,
      height: -2,
    },
    shadowRadius: 3,
    shadowOpacity: 0.2,
    shadowColor: 'black',
    textAlign: 'center',
  },
  headingz: {
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: 15,
    color: '#333333',
  },
  pickContainerz: {
    flex:1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  pickWrapperz: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 10,
  },
  circles: {
    height: 33,
    borderRadius: 30,
    width: 33,
    backgroundColor: '#A2A1A1',
    alignItems: 'center',
    justifyContent: 'center',
  },
  circlesTwo: {
    height: 55,
    borderRadius: 0,
    width: 55,
    backgroundColor: '#ffffff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  circlesTop: {
    height: 30,
    borderRadius: 0,
    width: 30,
    backgroundColor: '#49C7E3',
    alignItems: 'center',
    justifyContent: 'center',
  },
  circlesThree: {
    height: 33,
    borderRadius: 30,
    width: 33,
    backgroundColor: '#A2A1A1',
    alignItems: 'center',
    justifyContent: 'center',
  },
  circleActivez: {
    backgroundColor: 'red',
  },
  arrowWinz: {
    width: 30,
    height: 30,
  },
  arrowDrawz: {
    width: 30,
    height: 30,
  },
  topIcons: {
    width: 22,
    height: 22,
  },
  buttonzRed: {
    height: 40,
    borderRadius: 4,
    width: 320,
    marginBottom: 30,
    backgroundColor: '#EE4C50',
    alignItems: 'center',
    justifyContent: 'center',
  }
});
