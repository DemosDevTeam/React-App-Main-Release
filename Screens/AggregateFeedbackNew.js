import React from 'react'
import { View, StyleSheet, Text, AsyncStorage, ScrollView } from 'react-native'
import { to } from '../components/util'
import { FeedItem } from '../components'
import firebaseApp from '../firebaseApp'


const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})

export default class AggregateFeedback extends React.Component {
  state = {
    articles: { }
  }

  constructor(props){
    super(props)
  }

  componentDidMount = async () => {
    console.log("inside of componentDidMount");
    // Fetch all recent articles
    let err, articles;
    [err, articles] = await to(this.fetchArticles());

    if (!articles && err) {
      throw new Error("Failed to fetch articles")
    }

    this.setState({ articles });
  }

  fetchArticles = async () => {
    console.log("Inside of fetchArticles");
    const user = await AsyncStorage.getItem('user');
    const database = firebaseApp.database();
    var articles = {};
    var articleTitles = [];
    var cities = [];

    console.log("getting all article titles");
    //get titles of all articles to display
    await database.ref('/Users/' + user + '/Reactions/').once('value').then(snap => {
      snap.forEach(child => {
        articleTitles.push(child.key);
      })
      return articleTitles;
    })

    console.log("getting user cities");
    //get cities that user is interested inspect
    await database.ref('/Users/' + user + '/cities/').once('value').then(snap => {
      snap.forEach(child => {
        cities.push(child.key);
      })
    })

    console.log("getting actual article");
    //get json associated with each articleTitle
    await database.ref('/videos/').once('value').then(snap => {
      //iterate thorugh cities, check cities that user subscribes to.
      snap.forEach(city => {
        if(cities.includes(city.key)){

          //iterate through videos for city and check if user provided feedback on those cities.
          city.forEach(video => {
            if(articleTitles.includes(video.key)){
              //set json object associated with this video to be video JSON
              //add city name to the json object we write
              articles[video.key] = video.val();
              articles[video.key].city = city.key;
            }
          })
        }
      })
    })

    return articles;

  }


  render() {
    const { articles } = this.state;

    let articlesJsx;

    // Empty check
    if (!articles || Object.entries(articles).length == 0) {
      console.log("inside of empty check for articles");
      articlesJsx = (
        <View>
        <Text>Oh no! The political sphere is ruminating... (No articles available)</Text>
        </View>
      )
    } else {
      console.log("passed the empty check");
      const { navigation } = this.props

      articlesJsx = Object.entries(articles).map(([articleId, article]) => (
        <FeedItem
        key={articleId}
        article={article}
        onPress={() => navigation.navigate('Article', {
          articleId,
          article: {
            'id': articleId,
            ...article
          }
        })}
        />
      ))
    }

    return (
      <ScrollView styles={{flex: 1, justifyContent: 'space-between'}}>
          {articlesJsx}
      </ScrollView>
    );
  }
}
