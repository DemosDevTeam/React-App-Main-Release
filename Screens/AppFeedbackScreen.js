import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Button,
  TouchableOpacity,
  TouchableHighlight,
  Image,
  TextInput,
  Alert,
  ScrollView,
  AsyncStorage,
} from 'react-native';
import firebaseApp from '../firebaseApp'

export default class AppFeedbackScreen extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      'feedback': ''
    }
  }



  storeFeedback = (feedback) => {
    this.state.feedback = feedback;
    console.log("value of feedback is");
    console.log(this.state.feedback);
  }

  submit = async () => {
    const user = await AsyncStorage.getItem('user');
    await firebaseApp.database().ref('/Users/' + user + '/App Feedback/').set(this.state.feedback);
    this.props.navigation.navigate('Settings');
  }

  render() {
    return (
      <ScrollView>
      <View style={{alignItems: 'center', justifyContent: 'center', marginTop: 20, marginLeft: 20, marginRight: 20}}>
      <Image
      style={{width: 120, height: 120, marginTop: 20, marginBottom: 20}}
      source={{uri: 'https://user-images.githubusercontent.com/18129905/35187343-734d21b4-fdf0-11e7-8799-761570dea412.png'}}
      />
      <View style={{height: 20}}></View>
      <Text>The main purpose of Acta is to increase the transparency and communication between you and your local government. We intend for this app to serve as an accessible, easy-to-use outlet to empower you with information and let your voice be heard by the decision-makers.
      </Text>
      </View>
      <View style={{height: 20}}></View>
      <View style={{alignItems: 'center', justifyContent: 'center', marginLeft: 20, marginRight: 20}}>
      <Text>We want your feedback! What do you like and dislike about the app? How can we improve it to fit your individual needs and support your voice?
      </Text>
      <View style={{height: 20}}></View>
        <TextInput
        placeholder='   Please add your feedback here'
        style={{marginLeft: 19, marginRight: 19, backgroundColor: '#DFDFE2', borderRadius: 3}}
        editable={true}
        multiline={true}
        onChangeText={this.storeFeedback}/>
        <View style={{height: 10}}></View>
        <View style={{marginLeft: 30, marginRight: 30, marginTop: 5, marginBottom: 5, borderRadius: 5, alignItems: 'center', backgroundColor: '#55CFE0', paddingTop: 3, paddingBottom: 3, paddingLeft: 3, paddingRight: 3}}>
        <TouchableOpacity onPress={this.submit}><Text>Submit feedback</Text></TouchableOpacity>
        </View>
        <View style={{height: 20}}></View>
        <Image
            style={{height: 45, width: 80}}
            source={{ uri: 'https://user-images.githubusercontent.com/18129905/43050902-6fc94544-8dde-11e8-94fd-9bdfa1df6ead.png'}}
        />
        </View>
      </ScrollView>
    )
  }


}
