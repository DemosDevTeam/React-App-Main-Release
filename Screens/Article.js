import React from 'react'

import { View } from 'react-native'

import { ArticleCard, CommentCard } from '../components'

export default class ArticleScreen extends React.Component {
    render() {
        return (
            <View>
                <ArticleCard />
                <CommentCard />
            </View>
        );
    }
}