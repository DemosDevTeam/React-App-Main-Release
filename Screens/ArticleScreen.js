import React from 'react'

import { View, ScrollView, ActivityIndicator } from 'react-native'

import { Article, CommentCard } from '../components'

import firebaseApp from '../firebaseApp'

class ArticleScreen extends React.Component {
    constructor(props) {
        super(props)

        this.state = {
            article: undefined
        }
    }

    async constructorDidMount() {
        const { navigation } = this.props

        const articleId = navigation.getParam('articleId', undefined);

        try {
            if (!articleId) {
                throw new Error("No article id provided")
            }

            const articleSnapshot = await this.fetchArticle(articleId);

            if (articleSnapshot.val()) {
                this.setState({ article: articleSnapshot.val() })                
            }
        } catch (error) {
            const {code, message} = error;

            console.error(code, message)
        }
    }

    fetchArticle = async (articleId) => {
        return await firebaseApp.database().ref(`articles/${articleId}`).once('value');
    }
    
    render() {
        const { article } = this.state

        const loading = (article) 
            ? <Article 
                title={article.title}
                videoUri={article.videoUri}
                content={article.content}
                source={article.source}
            /> 
            : <ActivityIndicator /> 

        return (
            <ScrollView>
                {loading}
            </ScrollView>
        );
    }
}

export default ArticleScreen;