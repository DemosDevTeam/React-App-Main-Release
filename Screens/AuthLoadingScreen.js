import React from 'react';

import { 
    ActivityIndicator,
    AsyncStorage,
    StatusBar,
    View
} from 'react-native'

class AuthLoadingScreen extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        const userEmail = this.fetchUserToken();
        
        this.props.navigation.navigate(userEmail ? 'App' : 'Auth');
    }

    fetchUserToken = async () => {
        try {
            const userToken = await AsyncStorage.getItem('userEmailHash');
            return userToken;
        } catch (error) {
            console.error(error);
        }
    }

    render() {
        return (
            <View>
                <ActivityIndicator />
                <StatusBar barStyle="default" />
            </View>
        )
    }
}

export default AuthLoadingScreen;