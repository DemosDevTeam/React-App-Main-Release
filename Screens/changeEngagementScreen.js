import React from 'react'
import { View, StyleSheet, Text, AsyncStorage, ScrollView, TouchableOpacity, Image } from 'react-native'
import { to } from '../components/util'
import FeedbackFeedItem from '../components/FeedbackFeedItem'
import firebaseApp from '../firebaseApp'

export default class changeEngagementScreen extends React.Component {
  state = {
    engagement: []
  }

  constructor(props) {
    super(props);
  }

  //On submit verify inputs and navigate to next registration screen
  submit = async () => {
    var res;
    AsyncStorage.getItem('user')
    .then(result => res = result)
    .then(async () => {
      if(this.state.engagement.length > 0) {
        await firebaseApp.database().ref("/Users/" + res + "/engagement/").remove();
        for(var i=0; i<this.state.engagement.length; i++){
          await firebaseApp.database().ref("/Users/" + res + "/engagement/").child(this.state.engagement[i]).set("");
        }
        this.props.navigation.navigate('Settings');
      } else {
        Alert.alert("Please make sure to select at least one engagement type!");
      }

    });
  }

  handleSelection = (text) => {
    if(this.state.engagement.indexOf(text) < 0){
      this.state.engagement.push(text);
    }
  }


  render() {
    console.disableYellowBox = true;
    return (
      <ScrollView>
      <View style={styles.container}>
      <Image
        style={{width: 100, height: 100, marginTop: 40, marginBottom: 20}}
        source={{uri: 'https://user-images.githubusercontent.com/18129905/35187343-734d21b4-fdf0-11e7-8799-761570dea412.png'}}
      />
        <Text style={{fontSize: 16}}>What do you want your engagement level to look like?</Text>
        <Text style={{fontSize: 15}}>(Choose as many as you would like)</Text>
        <View style={styles.space}></View>
        <View style={styles.space}></View>
      </View>
        <View style={styles.space}></View>
        <View style={styles.container}>
        <View style={styles.buttonz2}>
          <TouchableOpacity onPress={() => this.handleSelection("online polls")}>
            <Text style={{fontSize: 15, textAlign: 'center'}}>Taking Online Polls</Text>
          </TouchableOpacity>
        </View>
        </View>

        <View style={styles.bspace}></View>
        <View style={styles.container}>
        <View style={styles.buttonz2}>
          <TouchableOpacity onPress={() => this.handleSelection("learn about local gov")}>
            <Text style={{fontSize: 15, textAlign: 'center'}}>Learning about Local Government</Text>
          </TouchableOpacity>
        </View>
        </View>
        <View style={styles.bspace}></View>
        <View style={styles.container}>
        <View style={styles.buttonz2}>
          <TouchableOpacity onPress={() => this.handleSelection("contact council")}>
            <Text style={{fontSize: 15, textAlign: 'center'}}>Contacting Your Council</Text>
          </TouchableOpacity>
        </View>
        </View>

        <View style={styles.bspace}></View>
        <View style={styles.container}>
        <View style={styles.buttonz2}>
          <TouchableOpacity onPress={() => this.handleSelection("meet with council")}>
            <Text style={{fontSize: 15, textAlign: 'center'}}>Meeting with Your Council</Text>
          </TouchableOpacity>
        </View>
        </View>

        <View style={styles.bspace}></View>
        <View style={styles.container}>
        <View style={styles.buttonz2}>
          <TouchableOpacity onPress={() => this.handleSelection("events")}>
            <Text style={{fontSize: 15, textAlign: 'center'}}>Attending an Event</Text>
          </TouchableOpacity>
        </View>
        </View>
        <View style={styles.space}></View>
        <View style={styles.space}></View>
        <View style={styles.space}></View>
        <View style={styles.container}>
        <View style={styles.buttonz}>
          <TouchableOpacity onPress={this.submit}>
            <Text style={{fontSize: 16, textAlign: 'center'}}>Submit and Continue</Text>
          </TouchableOpacity>
        </View>
        </View>
      </ScrollView>
    )
  }
}

var styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  images: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    flex: 0,
    marginBottom: 5,
    marginTop: 10,
  },
  space: {
    height: 2,
  },
  bspace: {
    height: 0,
  },
  space2: {
    height: 20,
  },
  userInputs: {
    marginTop: 15,
    marginBottom: 15,
  },
  buttonz: {
    height: 40,
    borderRadius: 4,
    width: 320,
    backgroundColor: '#49C7E3',
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonz2: {
    height: 29,
    borderRadius: 4,
    width: 320,
    marginTop: 3,
    marginBottom: 3,
    backgroundColor: '#c9c9c9',
    alignItems: 'center',
    justifyContent: 'center',
}
});
