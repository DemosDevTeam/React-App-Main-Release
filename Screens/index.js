import SplashScreen from './Home'
import LoginScreen from './Login'
import MainFeedScreen from './MainFeedNew'
import CouncilScreen from './CouncilScreen'
import ArticleScreen from './ArticleScreen'
import RegistrationScreen from './RegistrationScreen1'

export {
    SplashScreen,
    MainFeedScreen,
    LoginScreen,
    CouncilScreen,
    ArticleScreen,
    RegistrationScreen
}