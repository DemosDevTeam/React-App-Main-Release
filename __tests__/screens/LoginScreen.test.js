import 'react-native'
import React from 'react'

import { LoginScreen } from '../../Screens'

import renderer from 'react-test-renderer'

jest.mock('../../firebaseApp')

test('render correctly', () => {
    const tree = renderer.create(<LoginScreen />).toJSON()

    expect(tree).toMatchSnapshot()
})