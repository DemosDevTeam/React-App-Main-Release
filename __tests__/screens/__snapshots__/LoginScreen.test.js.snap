// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://user-images.githubusercontent.com/18129905/35187343-734d21b4-fdf0-11e7-8799-761570dea412.png",
      }
    }
    style={
      Object {
        "height": 200,
        "marginBottom": 40,
        "marginTop": 75,
        "width": 200,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
  />
  <View
    style={
      Object {
        "height": 20,
      }
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#49C7E3",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 4,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    or
  </Text>
  <View
    style={
      Object {
        "height": 4,
      }
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#3B5998",
            },
          ]
        }
      >
        Login with Facebook
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 4,
      }
    }
  />
  <View
    style={
      Object {
        "height": 2,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#A2A1A1",
        "fontSize": 10,
        "textAlign": "center",
      }
    }
  >
    ______________________________________________________________________
  </Text>
  <View
    style={
      Object {
        "height": 4,
      }
    }
  />
  <View
    style={
      Object {
        "height": 4,
      }
    }
  />
  <View
    style={
      Object {
        "height": 4,
      }
    }
  />
  <View
    style={
      Object {
        "height": 4,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    New here?
  </Text>
  <View
    style={
      Object {
        "height": 2,
      }
    }
  />
  <View
    style={
      Object {
        "height": 2,
      }
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#EE4C50",
            },
          ]
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
